app/Http/Controllers/Api/ItemTypeController.php

```diff
@@ -0,0 +1,112 @@
+<?php namespace App\Http\Controllers\Api; 
+
+use App\Http\Requests;
+use App\Http\Controllers\Controller;
+
+use App\ItemType;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Response;
+use Illuminate\Database\Eloquent\ModelNotFoundException;
+
+class ItemTypeController extends Controller {
+
+	/**
+	 * Return inventory types 
+	 *
+	 * @return Inventory item types
+	 */
+	public function index()
+	{
+		$httpContents = '';
+		$httpStatus = 200;
+		$itemTypes = null;
+
+		$itemTypes = ItemType::all();
+		
+		if(is_null($itemTypes))
+		{
+			$httpContents = 'Object is null';
+			$httpStatus = 404;
+		}
+		else
+		{
+			// Check if content is empty
+			if(empty($itemTypes))
+			{
+				$httpContents = 'No item types available';
+			}
+			else
+			{
+				$httpContents = $itemTypes;
+			}
+		}
+
```

Comment:
We really don't need to check whether the result of retrieving all `ItemType`s is null or empty.

Querying `ItemType::all()` should always return a list of item types. It might be empty, but that's OK. This is a collection, so if it's empty, that's not a failure case.

We want the client to be able to trust that what we send back is valid JSON that it can parse and use. If we return the string `'No item types available'`, we're breaking that rule and giving the client something it can't use programmatically.